### PR TITLE
8300586: Refactor code examples to use @snippet in java.text.Collator

### PR DESCRIPTION
**Pull request by justin-curtis-lu**
This PR implements _JEP 413: Code Snippets in Java API Documentation_ for [java.text.Collator](https://docs.oracle.com/en/java/javase/19/docs/api/java.base/java/text/Collator.html).

Code examples using \<pre> ... \</pre> blocks are replaced with the @ snippet syntax where applicable.

Additionally, the code examples have been slightly tweaked to increase readability.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8300586](https://bugs.openjdk.org/browse/JDK-8300586): Refactor code examples to use @snippet in java.text.Collator

### Reviewers
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**)
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**)
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12080/head:pull/12080` \
`$ git checkout pull/12080`

Update a local copy of the PR: \
`$ git checkout pull/12080` \
`$ git pull https://git.openjdk.org/jdk pull/12080/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12080`

View PR using the GUI difftool: \
`$ git pr show -t 12080`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12080.diff">https://git.openjdk.org/jdk/pull/12080.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/java/text/Collator.java [hunk 1]
```diff
@@ -72,14 +72,15 @@
...
  * Collator myCollator = Collator.getInstance();
- * if( myCollator.compare("abc", "ABC") < 0 )
+ * if (myCollator.compare("abc", "ABC") < 0) {
  *     System.out.println("abc is less than ABC");
- * else
+ * } else {
  *     System.out.println("abc is greater than or equal to ABC");
- * }</pre>
+ * }
+ * }
```
Comment (irisclark) [inline src/java.base/share/classes/java/text/Collator.java:83]: I don't think you need a second '}' here.

**Review by irisclark — APPROVED** [review]

File: src/java.base/share/classes/java/text/Collator.java (hunk 1 above)
Comment (irisclark) [reply src/java.base/share/classes/java/text/Collator.java:83]: I see that now!  I didn't look up far enough.  Thanks!

Comment (justin-curtis-lu) [reply src/java.base/share/classes/java/text/Collator.java:83]: No worries, thank you for reviewing!

**Review by LanceAndersen — APPROVED** [review]

**Review by naotoj — APPROVED** [review]

**Pull request closed without merging**